plugins/parcelport/libfabric/libfabric_region_provider.hpp

```diff
@@ -31,7 +34,13 @@ namespace libfabric
 
         template <typename... Args>
         static int register_memory(Args &&... args) {
-            return fi_mr_reg(std::forward<Args>(args)...);
+            int res = fi_mr_reg(std::forward<Args>(args)...);
+            if (res == -FI_ENOKEY)
+                std::cerr << "key already in use\n";
+            if (res == -FI_EKEYREJECTED)
+                std::cerr << "key not available\n";
+
+            return res;
```

Comment:
Have you been getting memory registration errors? I had to reuse the same keys to make things work (with the psm provider) - if this is broken, then I may need to do some bigger changes in the memory handling.